src/main/java/org/broadinstitute/consent/http/service/dao/DraftServiceDAO.java

```diff
@@ -0,0 +1,156 @@
+    DraftInterface draft;
+    try {
+      draft = findDraftByDraftUUID(draftUUID);
+    } catch (SQLException e) {
+      throw new NotFoundException(
+          String.format("Draft with UUID %s not found.", draftUUID.toString()));
+    }
+    if (Objects.isNull(draft)) {
+      throw new NotFoundException(
+          String.format("Draft with UUID %s not found.", draftUUID.toString()));
+    }
+    if (!user.getUserId().equals(draft.getCreateUser().getUserId()) && !user.hasUserRole(
+        UserRoles.ADMIN)) {
+      throw new NotAuthorizedException("User not authorized to modify resource.");
+    }
+    return draft;
+  }
+
+  public void deleteDraftsByUser(User user) {
+    Set<DraftInterface> userDrafts = findDraftsForUser(user);
+    for (DraftInterface draft : userDrafts) {
+      deleteDraft(draft, user);
+    }
+  }
+
+  public Set<DraftSummary> findDraftSummariesForUser(User user) {
+    return draftDAO.findDraftSummariesByUserId(user.getUserId());
+  }
+
+  public Set<DraftInterface> findDraftsForUser(User user) {
+    return draftDAO.findDraftsByUserId(user.getUserId());
+  }
+
+  private DraftInterface findDraftByDraftUUID(
+      UUID draftUUID) throws SQLException {
+    return draftDAO.findDraftById(draftUUID);
+  }
+
+  public List<FileStorageObject> addAttachments(DraftInterface draft, User user,
+      Map<String, FormDataBodyPart> files) throws SQLException {
+    List<FileStorageObject> storedFiles = draftFileStorageServiceDAO.storeDraftFiles(draft.getUUID(), user, files);
+    draftDAO.updateDraftByDraftUUID(draft.getUUID(),
+        new Date().toInstant(), user.getUserId());
+    return storedFiles;
+  }
+
+  public void deleteDraftAttachment(DraftInterface draft, User user, Integer fileId)
+      throws SQLException {
+    Optional<FileStorageObject> fileStorageObjectToDelete = draft.getStoredFiles().stream()
+        .filter(fileStorageObject -> fileStorageObject.getFileStorageObjectId().equals(fileId))
+        .findFirst();
+    if (fileStorageObjectToDelete.isPresent()) {
+      draftFileStorageServiceDAO.deleteStoredFile(fileStorageObjectToDelete.get(), user);
+      draftDAO.updateDraftByDraftUUID(draft.getUUID(),
+          new Date().toInstant(), user.getUserId());
+    } else {
+      throw new NotFoundException(
+          String.format("Draft attachment is not found.  Draft: %s, Attachment: %d",
+              draft.getUUID(), fileId));
+    }
+  }
+
+  public void deleteDraft(DraftInterface draft, User user)
+      throws RuntimeException {
+    jdbi.useHandle(handle -> {
+      try {
+        handle.useTransaction(handler -> {
+          draftDAO.deleteDraftByUUIDList(List.of(draft.getUUID()));
+          draft.getStoredFiles().forEach(fileStorageObject -> {
+            try {
+              draftFileStorageServiceDAO.deleteStoredFile(fileStorageObject, user);
+            } catch (SQLException e) {
+              throw new RuntimeException(e);
+            }
+          });
```

Comment:
It looks like the purpose of this try/catch is to allow the use of `forEach()`. If so, I think it would be clearer to write this as a loop and avoid `forEach()`.
```suggestion
          for (FileStorageObject fileStorageObject : draft.getStoredFiles()) {
            draftFileStorageServiceDAO.deleteStoredFile(fileStorageObject, user);
          }
```